canvas.go

```diff
@@ -269,28 +269,36 @@ func (c *Canvas) servicesGroup(canvas *svg.SVG) {
 	}
 }
 
-// Marshal renders the SVG to the given io.Writer
-func (c *Canvas) Marshal(w io.Writer) {
-	width, height := c.layout()
+// Compute the scale.
+func (c *Canvas) computeScale(width, height int) float32 {
 	scale := float32(1)
 	if height > maxHeight {
 		scale = maxHeight / float32(height)
 	}
 	if float32(width)*scale > maxWidth {
-		scale = maxHeight / float32(height)
+		scale = maxWidth / float32(width)
 	}
 
+	return scale
+}
+
+// Marshal renders the SVG to the given io.Writer.
+func (c *Canvas) Marshal(w io.Writer) {
+
 	// TODO check write errors and return an error from
 	// Marshal if the write fails. The svg package does not
-	// itself check or return write errors; a possible workaround
+	// itself check or return write errors; a possible work-around
 	// is to wrap the writer in a custom writer that panics
 	// on error, and catch the panic here.
+	width, height := c.layout()
+	scale := c.computeScale(width, height)
 
 	canvas := svg.New(w)
 	canvas.Start(
 		width,
 		height,
-		`style="font-family:Ubuntu, sans-serif;"`,
+		fmt.Sprintf(`style="font-family:Ubuntu, sans-serif;" viewBox="0 0 %d %d"`,
+			width, height),
```

Comment:
Correct. Viewbox is a weird attribute that defines both the size of the view portal and also whether or not the image can be scaled instead of clipped.